test/server.cpp

```diff
@@ -194,14 +194,14 @@ ResourceCollection all200Resources()
 TEST_F(ServerTest, 200)
 {
   for ( const Resource& res : all200Resources() )
-    EXPECT_EQ(200, zfs1_->GET(res.url)->status) << "res.url: " << res.url;
+    EXPECT_GE(204, zfs1_->GET(res.url)->status) << "res.url: " << res.url;
```

Comment:
That's not a good check. It allows for any response code <= 204, and doesn't validate which status is returned for which request.

A good test must possess the following properties:

1. It must pass if the code is correct
2. It must fail if the code is not correct

It is a typical mistake to forget about item 2, since people usually feel happy when they observe passing tests. As a result sometimes tests are created that have no chance of failing, or gradually degrade toward that state. In this particular case the test has degraded - the server may return status code 123 but the test will not catch that.   

The same comment applies to all your other changes in `tests/server.cpp`

Reply:
ok I didn't notice that one I would try to make changes around that, thanks if you could give me any idea of sort of to help please feel free to tell

Reply:
To the best of my understanding, the root cause for the server tests failing because of your changes is that the tested URLs contain at least one search query that returns an empty result. You can modify those URLs so that the search results are non-empty. And then you can add an additional (separate) test case for your enhancement.

Reply:
Ok I was thinking about removing EXPECT_EQ macro for 200 status code and implementing it with 2 OR statements in condition but if that is an ill practice I will go by your idea thanks.